projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -17,3 +23,48 @@ RADIUS::RADIUS()
 {
 }
 
+int RADIUS::Start()
+{
+    m_thread = std::jthread([this](auto token){ Run(std::move(token)); });
+    return 0;
+}
+
+int RADIUS::Stop()
+{
+    std::lock_guard lock(m_mutex);
+
+    if (!m_thread.joinable())
+        return 0;
+
+    m_thread.request_stop();
+
+
+    if (isRunning)
+        m_thread.detach();
+    else
+        m_thread.join();
+
+    return 0;
+}
+
+int RADIUS::Run(std::stop_token token)
+{
+    std::lock_guard lock(m_mutex);
```

Comment:
This lock has a function-level scope. It will be released when the function finishes. It means that no one can access `isRunning` while the function is running. You should hold locks only when you read or write a variable, `isRunning` in particular, and release them as soon as possible - after reading or writing.